CMakeLists.txt

```diff
@@ -796,6 +796,22 @@ if(EXECUTORCH_BUILD_PYBIND)
 endif()
 
 if(EXECUTORCH_BUILD_WASM)
+  if(NOT EXECUTORCH_BUILD_EXTENSION_DATA_LOADER)
```

Comment:
@larryliu0820  I feel like this pattern used to be really common in our cmake but now taking a look I dont see it anywhere else except pybind. How does something like Module guarantee that data loader is built now?

Reply:
I guess for now @Conarnar just follow the existing pattern and shift to the requires options thing